### PR TITLE
Remove mobile_app from generated Lovelace

### LINKED ISSUE #6742: Generated UI should not include mobile app entities
## The request

When you set up a mobile phone (as we suggest during onboarding), your home screen is bombarded with entities. That's not really helpful. 

The generated UI should not include any mobile app entities. The person entity that is linked to the mobile app will still show the status of the phone (if it's home or not).

## The alternatives

Overwhelm new users.

### PR DESCRIPTION
**Pull request by ludeeus**
## Breaking change

Entities that are generated from mobile apps with the `mobile_app` integration is now hidden in the generated Lovelace view.
If you want to continue to display those you need to take control over your view with the 3 dots in the top right corner of the Lovelace screen.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #6742
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/common/generate-lovelace-config.ts [hunk 1]
```diff
@@ -413,6 +413,15 @@ export const generateLovelaceConfigFromData = async (
     viewEntities.length === 0 ||
     viewEntities[0].entity_id !== DEFAULT_VIEW_ENTITY_ID
   ) {
+    const hiddenPlatforms = ["mobile_app"];
```
Comment (bramkragten) [inline src/panels/lovelace/common/generate-lovelace-config.ts:416]: Move this to the top of the file

File: src/panels/lovelace/common/generate-lovelace-config.ts [hunk 2]
```diff
@@ -413,6 +413,15 @@ export const generateLovelaceConfigFromData = async (
...
+    const hidddenEntitites = entityEntries
+      .filter((entry) => hiddenPlatforms.includes(entry.platform))
+      .map((entry) => entry.entity_id);
+
+    hidddenEntitites.forEach((entity) => {
+      delete entities[entity];
+    });
```
Comment (bramkragten) [inline src/panels/lovelace/common/generate-lovelace-config.ts:423]:
```suggestion
    const entities = entityEntries
      .filter((entry) => hiddenPlatforms.includes(entry.platform));
```

Comment (bramkragten) [reply src/panels/lovelace/common/generate-lovelace-config.ts:423]: Won't this just remove them from the area? It is still in states?

Comment (ludeeus) [reply src/panels/lovelace/common/generate-lovelace-config.ts:423]: Yes, that change will not work, but I'll see if I can make my current implementation simpler

File: src/panels/lovelace/common/generate-lovelace-config.ts [hunk 3]
```diff
@@ -63,6 +63,8 @@ const HIDE_DOMAIN = new Set([
   "zone",
 ]);
 
+const HIDE_PLATFORM = ["mobile_app"];
```
Comment (bramkragten) [inline src/panels/lovelace/common/generate-lovelace-config.ts:66]:
```suggestion
const HIDE_PLATFORM = new Set(["mobile_app"]);
```

File: src/panels/lovelace/common/generate-lovelace-config.ts [hunk 4]
```diff
@@ -206,11 +208,21 @@ export const computeCards = (
...
 
-const computeDefaultViewStates = (entities: HassEntities): HassEntities => {
+const computeDefaultViewStates = (
+  entities: HassEntities,
+  entityEntries: EntityRegistryEntry[]
+): HassEntities => {
   const states = {};
+  const hiddenEntities = entityEntries
+    .filter((entry) => HIDE_PLATFORM.includes(entry.platform))
+    .map((entry) => entry.entity_id);
```
Comment (bramkragten) [inline src/panels/lovelace/common/generate-lovelace-config.ts]:
```suggestion
  const hiddenEntities = new Set(
    entityEntries
      .filter((entry) => HIDE_PLATFORM.has(entry.platform))
      .map((entry) => entry.entity_id)
  );
```

File: src/panels/lovelace/common/generate-lovelace-config.ts [hunk 5]
```diff
@@ -206,11 +208,21 @@ export const computeCards = (
...
 };
 
-const computeDefaultViewStates = (entities: HassEntities): HassEntities => {
+const computeDefaultViewStates = (
+  entities: HassEntities,
+  entityEntries: EntityRegistryEntry[]
+): HassEntities => {
   const states = {};
+  const hiddenEntities = entityEntries
+    .filter((entry) => HIDE_PLATFORM.includes(entry.platform))
```
Comment (zsarnett) [inline src/panels/lovelace/common/generate-lovelace-config.ts]:
```suggestion
    .filter((entry) => HIDE_PLATFORM.has(entry.platform))
```

Does Set use has?

File: src/panels/lovelace/common/generate-lovelace-config.ts [hunk 6]
```diff
@@ -206,11 +208,21 @@ export const computeCards = (
...
+    .map((entry) => entry.entity_id);
+
   Object.keys(entities).forEach((entityId) => {
     const stateObj = entities[entityId];
-    if (!HIDE_DOMAIN.has(computeStateDomain(stateObj))) {
+    if (
+      !HIDE_DOMAIN.has(computeStateDomain(stateObj)) &&
+      !hiddenEntities.includes(entityId)
```
Comment (bramkragten) [inline src/panels/lovelace/common/generate-lovelace-config.ts:224]:
```suggestion
      !hiddenEntities.has(stateObj.entity_id) &&

```

File: src/panels/lovelace/common/generate-lovelace-config.ts (hunk 5 above)
Comment (zsarnett) [reply src/panels/lovelace/common/generate-lovelace-config.ts]: You fixed this as I commented :)

Comment (bramkragten) [reply src/panels/lovelace/common/generate-lovelace-config.ts]: Yes

**Review by bramkragten — APPROVED** [review]

**Comment by ludeeus** [comment]
Would this be a breaking change?

**Comment by bramkragten** [comment]
Uuuuuhhhh... probably? We didnt mark it as breaking change before I think

**Comment by ludeeus** [comment]
Someone remind me about this when we are getting ready for 0.116 :D

**Pull request merged**